### ISSUE TITLE
Trackpad driver with multitouch gestures

### ISSUE THREAD

**Issue by beautifulsole** [Kaijun/Acer-V5-573G-Hackintosh#24]
Hello.  I recently bought one of these laptops - the one with the i7 and 750M graphics, and immediately got it hackintoshed.

I found a driver that works extremely well for our laptops, refer to this page: http://forum.osxlatitude.com/index.php?/topic/1948-elan-focaltech-and-synaptics-smart-touchpad-driver-mac-os-x/

It features multitouch gesture capabilities, including double-finger scroll, multifinger swipe and pinch, edge swipe, continuous scroll, and throws in keyboard remapping too, for producing that Mac effect of triggering F-key events without having to press the Fn button.

I'm extremely impressed by its functionality, though I am having a few issues with it, although the major one is that the trackpad and keyboard refuse to work if I make my laptop sleep, or if the screen turns off automatically, or if I close the lid without letting it sleep.  Funnily enough, it keeps functioning if I switch off the screen using Fn+F6 (or simply F6) as I've configured it.

Also, I'd like to ask a couple questions to y'all about your experience with this laptop: do you have a problem with low audio as compared to Windows?  Also, I can also report the issue Kaijun brought up earlier, in that sometimes it refuses to shut down and has to be shut down forcefully.  I'm hoping there's a fix for it soon.

Anyway, enjoy the new gestures!

**Comment by Ace73** [comment]
Hi,
Thanks for finding the driver. Could you upload it, as I don't want to sign up for an account on OSXLatitude.

**Comment by beautifulsole** [comment]
Here you go: http://s000.tinyupload.com/index.php?file_id=60287423623855790224

**Comment by Majkwin** [comment]
Is it working without any problems? What has changed comparing to ordinary
VoodooPS2 driver? As far as I know, voodoo enables keyboard as well as
trackpad, so we can not delete it to install this kext right? Or is it
enables keyboard as well as gestures?

2015-11-10 13:15 GMT+01:00 beautifulsole <notifications@github.com>:

**Comment by beautifulsole** [comment]
There are a few problems that I've commented with in the main thread of the forum, check out my post.  The deal-breaking issue was when the driver would fail on screen off/sleep, but I managed to find a way to work around that, as mentioned later on the same thread.  Check it out and give it a shot.

And it's a complete PS/2 kext, so keyboard works flawlessly, too.  Better, in fact; I can now use my Fn keys for Macbook functionality without pressing Fn, so, for example, I press F12 for Vol+, and so on.

Also, off topic, does anyone else have a problem with 2-finger tap to right click on Windows?  :P

**Comment by Ace73** [comment]
@beautifulsole, no, I don't have a problem with using 2-fingers to right click on Windows. Have you installed the ELAN driver?

**Comment by beautifulsole** [comment]
I have.  I should probably explain: I can double tap to right click on Mac, naturally, just fine.  On Windows, double-tap to right click doesn't work as naturally as one would hope.  My fingers have to touch the pad at EXACTLY the same time, and leave the pad EXACTLY at the same time, for the double tap to trigger.  I have no problems with double-finger scrolling, or any other "gestures" in the Elan controls.

Can you link me to the driver version you're using?  Perhaps I'm using the wrong driver version.  Thanks.

Also, did you test out the kext mentioned above?

**Comment by Ace73** [comment]
To answer your first part - My trackpad works like a charm. Download the driver from: http://global-download.acer.com/GDFiles/Driver/TouchPad/TouchPad_ELANTECH_15.6.0.3_W10x64_A.zip?acerid=635735828894598338&Step1=ULTRA-THIN&Step2=ASPIRE&Step3=ASPIRE%20V5-573G&OS=ALL&LC=en&BC=ACER&SC=AAP_7

To answer your second part - Yeah,  I am dualbooting Mac and Windows. The partition scheme, though, is GUID Partition Table [GPT] and not MBR. What do you mean by the installer saying your partition would become inaccessible? Where you on an MBR partition scheme? A similar problem did arise when I tried to hackintosh my desktop in like 2012 and then found out that the Mac installer formatted my hard drive completely as it was in MBR. MBR and Hackintosh don't play well together. I feel the best think to do [as you did] is to use UEFI mode, use GPT, install Windows and then install Mac OS X. I don't see why you couldn't directly install El Capitan. I currently am planning(not sure, though, as I do have a working install, only unable to boot into it) to format my Mac partition, install Mac OS X El Capitan directly [instead of updating] and removing and reinstalling Clover, as I believe updating Clover may have triggered the problems encountered earlier. You said you didn't want to screw things up, right? If you are on MBR, you could take a system image backup of Windows and restore it. I really feel that if you are using Legacy, you should switch over to UEFI.

**Comment by beautifulsole** [comment]
@Ace73 I tried the driver you linked; no dice.  I'm thinking now that it's a hardware related issue.  This really doesn't make much sense, though, as the trackpad works almost flawlessly in Mac.  At least the double-finger tap to right click does on Mac as it should.  This is frustrating, to say the least.  Ugh.

And to answer your second question, I'm using a pure UEFI setup, with my drive partitioned using GPT.  What I did was that I installed Mac first and Windows second, because I read somewhere that Mac doesn't like drives partitioned by Windows.  I don't want to mess with my currently-working setup though, and I don't have access to a backup disk.  I suppose the best thing to do now is to wait and cross my fingers and try to update to El Capitan as soon as I can boot into the installer and hope nothing goes wrong.  Speaking of which, have you managed to boot into the El Capitan installer?  As far as I remember, we seemed to have similar problems getting into the installer.
